decode.go

```diff
@@ -30,8 +32,8 @@ func decodedLen(src []byte) (blockLen, headerLen int, err error) {
 	if n <= 0 {
 		return 0, 0, ErrCorrupt
 	}
-	if uint64(int(v)) != v {
-		return 0, 0, errors.New("snappy: decoded block is too large")
+	if uint64(uint32(v)) != v {
```

Comment:
An easier and more explicit check is

if v >= 0xffffffff {

Still, there's a number of possible checks we could do here.

One is whether the reported v is <= 1<<32-1 as per https://github.com/google/snappy/blob/master/format_description.txt. The code prior to this patch does not check for this, and you do check for this now. This check is a property of the bytes on the wire, not the GOARCH.

Two is whether that value is representable as an int, the same type used to represent the length of a slice. This check is that the code prior to this patch attempts to do, and you no longer check for this: 1<<32-1 is a valid length according to the format, but returning that as an int is not do-able on GOARCH=arm. We should still check for this.

Related to that is whether int(v) can somehow be negative even if the above checks pass. Compare:
http://play.golang.org/p/-rQWyQ0wUU
http://play.golang.org/p/ba_TIO0bax